tests/callbacks/test_model_checkpoint.py

```diff
@@ -93,3 +99,24 @@ def test_model_checkpoint_no_extraneous_invocations(tmpdir):
     )
     result = trainer.fit(model)
     assert 1 == result
+
+
+def test_model_checkpoint_save_last_checkpoint_contents(tmpdir):
+    """ Tests that the checkpoint saved as 'last.ckpt' contains the latest information. """
+    seed_everything(100)
+    model = EvalModelTemplate()
+    num_epochs = 3
+    model_checkpoint = ModelCheckpoint(filepath=tmpdir, save_top_k=num_epochs, save_last=True)
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        early_stop_callback=False,
+        checkpoint_callback=model_checkpoint,
+        max_epochs=num_epochs,
+    )
+    trainer.fit(model)
+    path_last_epoch = model_checkpoint.format_checkpoint_name(num_epochs - 1, {})
+    path_last = tmpdir / CHECKPOINT_NAME_LAST
+    ckpt_last_epoch = torch.load(str(path_last_epoch))
```

Comment:
let's add assert that the checkpoint is not the same
aka `path_last_epoch != path_last`

Reply:
good idea!!!

Reply:
done :)